nuxeo-features/nuxeo-platform-task/nuxeo-platform-task-core/src/main/java/org/nuxeo/ecm/platform/task/core/service/TaskServiceImpl.java

```diff
@@ -174,8 +177,11 @@ public String rejectTask(CoreSession coreSession, NuxeoPrincipal principal, Task
     public String endTask(CoreSession coreSession, NuxeoPrincipal principal, Task task, String comment,
             String eventName, boolean isValidated) {
 
-        if (!canEndTask(principal, task)) {
-            throw new NuxeoException(String.format("User with id '%s' cannot end this task", principal.getName()));
+        if (!isTaskActive(task)) {
+            throw new ConcurrentUpdateException(String.format("The task '%s' is not active", task.getName()));
+        } else if (!canEndTask(principal, task)) {
```

Comment:
I would switch the two cases because if user doesn't have permission to end the task, he still can know if the task is active where he shouldn't. Wdyt?